pkg/kubectl/scale.go

```diff
@@ -81,6 +82,17 @@ func (precondition *ScalePrecondition) Validate(controller *api.ReplicationContr
 	return nil
 }
 
+// ValidateJob ensures that the preconditions match.  Returns nil if they are valid, an error otherwise
+func (precondition *ScalePrecondition) ValidateJob(job *experimental.Job) error {
+	if precondition.Size != -1 && *job.Spec.Parallelism != precondition.Size {
```

Comment:
If the field is required internally, we typically do not nil check it.  Although in this case, if parallelism could *ever* be nil, then you should do the check.